pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -299,33 +280,135 @@ def build_model(
                 name="mu", var=mu_var, dims=model_config["mu"]["dims"]
             )
 
-            pm.Normal(
-                name="likelihood",
-                mu=mu,
-                sigma=sigma,
-                observed=target_,
-                dims=model_config["likelihood"]["dims"],
-            )
+            likelihood = self.create_likelihood(self.model_config, target_, mu)
+
+    def create_priors_from_config(self, model_config):
+        priors, dimensions = {}, {"channel": len(self.channel_columns), "control": len(self.control_columns)}
+        stacked_priors = {}
+    
+        positive_params = {"intercept", "beta_channel", "alpha", "lam", "sigma"}  # Set of params that need positive=True
+
+        for param, config in model_config.items():
+            if param == "likelihood": continue
+
+            prior_type = config.get("type")
+            if prior_type is not None:
+               
+                # Initial value based on parameter name
+                is_positive = param in positive_params
+
+                # Override if the config explicitly sets the 'positive' key
+                if 'positive' in config:
+                    is_positive = config.get('positive')
+
+                if prior_type == "tvp":
+                    if param in ["intercept", "lam", "alpha", "sigma"]:
+                        priors[param] = self.gp_wrapper(name=param, X=np.arange(len(self.X[self.date_column]))[:, None], config=config, positive=is_positive)
+                        continue
+
+                    length = dimensions.get(config.get("dims", [None, None])[1], 1)
```

Comment:
needlessly terse. can be simplified to 

`dimensions.get(config.get("dims", None), 1)`

Reply:
my apologies, my suggesion won't work and this is actually decent code.